lib/types/config/ServicesConfiguration.ts

```diff
@@ -0,0 +1,298 @@
+import { JSONObject } from '../../../index';
+
+export type ServicesConfiguration = {
```

Comment:
Since there are only 3 external service configurations described under the `services` key and this will be modified to support different tools (Postgres as a possible replacement for Elasticsearch as storageEngine, ...) maybe it is time to prepare the extensibility of this part of the configuration. 

For example by declaring an interface for each of these three keys (storageEngine, internalCache, memoryStorage) and use them to implement future services compatibility. 

It would be enough to make the next specific configurations inherit from these interfaces and thus avoid the problems of key collisions, silent omission of some too specific configuration fields etc... I don't know how interesting the ROI is but I think we should discuss it 🙂

Reply:
For example:
```js
export interface IStorageEngineConfiguration { /* ... */ }
export interface IInternalCacheConfiguration { /* ... */ }
export interface IMemoryStorageConfiguration { /* ... */ }
```

Reply:
The types for services should be like this:

```
type StorageService = StorageServiceElasticsearch // in the futur we will have: StorageServiceElasticsearch | StorageServicePostgre

type StorageServiceElasticsearch = {
  backend: 'elasticsearch',
  aliases: string[],
  // etc
}
```